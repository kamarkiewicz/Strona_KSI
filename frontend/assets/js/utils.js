
// Access image field from backend
function getImage (image) {
  if ((image || {}).path) {
    // path may be relative URL
    let isAbsolute = /^(?:[a-z]+:)?\/\//i.test(image.path)
    image.path = (isAbsolute ? '' : process.env.API_URL) + image.path
  }
  return image
}

export { getImage };
