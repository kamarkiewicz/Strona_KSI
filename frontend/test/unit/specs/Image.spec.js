import _ from 'lodash'
import { Image } from '~/assets/js/utils'

describe.each([
  ///
  /// samples from News collection (fieldtype Image)
  ///

  [], // empty array

  {
    "path": "https://picsum.photos/505/295?image=1024"
  },

  {
    "path": "/storage/uploads/2018/04/26/5ae258ff2ac8d295.jpg",
    "title": "295.jpg",
    "mime": "image/jpeg",
    "description": "",
    "tags": [],
    "size": 17100,
    "image": true,
    "video": false,
    "audio": false,
    "archive": false,
    "document": false,
    "code": false,
    "created": 1524783359,
    "modified": 1524783359,
    "_by": "5ae2576cbbb96doc1618355576",
    "width": 505,
    "height": 295,
    "colors": [
        "1c1f1c",
        "cad6db",
        "6d6a66",
        "8b9496",
        "848c94"
    ],
    "_id": "5ae25a637a783doc1895663978"
  },

  ///
  /// samples from Galleries collection (fieldtype Gallery)
  ///

  {
    "meta": {
        "title": "",
        "asset": "5ae57e29dc5c7doc1474185002"
    },
    "path": "/storage/uploads/2018/04/29/5ae57e29cc7981035.jpg"
  },

])('Image object', (sample) => {
  let originalProcessEnv

  beforeEach(() => {
    originalProcessEnv = _.cloneDeep(process.env)
  })

  afterEach(() => {
    process.env = _.cloneDeep(originalProcessEnv)
  })

  it('constructs properly', () => {
    process.env.API_URL = 'https://www.example.com/api'
    expect(new Image(sample)).toMatchSnapshot()
  })

})

