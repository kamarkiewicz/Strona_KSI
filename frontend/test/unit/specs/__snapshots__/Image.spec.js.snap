// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image object constructs properly 1`] = `Image {}`;

exports[`Image object constructs properly 2`] = `
Image {
  "src": "https://picsum.photos/505/295?image=1024",
  "title": "Featured image",
}
`;

exports[`Image object constructs properly 3`] = `
Image {
  "height": 295,
  "src": "https://www.example.com/api/storage/uploads/2018/04/26/5ae258ff2ac8d295.jpg",
  "title": "295.jpg",
  "width": 505,
}
`;

exports[`Image object constructs properly 4`] = `
Image {
  "src": "https://www.example.com/api/storage/uploads/2018/04/29/5ae57e29cc7981035.jpg",
  "title": "5ae57e29dc5c7doc1474185002",
}
`;

exports[`Image object constructs properly 5`] = `Image {}`;
